### PR TITLE
Move locale code out of tm, into _config

### PR DESCRIPTION
**Pull request by jbrockmendel**
As discussed last week, one of ugliest runtime imports in `tslibs` is of `tm` for `tm.set_locale`.  This moves `set_locale` and a couple of other locale-related functions out of `tm`.

As discussed in #25162, #25203, #25613, this moves the affected functions to a new directory `pandas/_config`, intended to be strictly "upstream" from the rest of pandas (potentially even made actually-upstream as suggested [here](https://github.com/pandas-dev/pandas/pull/25176#issuecomment-461209852)).

Following this and #25613, a bunch of other cleanup/simplification becomes feasible.

### CODE REVIEW

**Review by WillAyd — COMMENTED** [review]
Not sure if I missed it in another thread but what is the issue with `get_locales` moving as well?

**Comment by jreback** [comment]
conceptually this looks good. some failing checks.

File: pandas/util/testing.py [hunk 1]
```diff
@@ -18,10 +17,13 @@
 import numpy as np
 from numpy.random import rand, randn
 
+from pandas._config.localization import (  # noqa:F401
```
[the remark this answers is not in the document]
Comment (jbrockmendel) [reply pandas/util/testing.py]: some of the imports are not used in this file, but are retained so that we can still use them as `tm.foo`

**Comment by jbrockmendel** [comment]
> what is the issue with get_locales moving as well?

`get_locales` depends on `pd.options.display.encoding`, so it can't be moved (in a way that keeps the dependency structure clean) until pd.core.config and parts of config_init are moved to _config.

**Comment by jbrockmendel** [comment]
CI checks fixed.

**Comment by jbrockmendel** [comment]
@jreback rebased following #25613.  Nice follow-up available after this.

File: pandas/util/testing.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jreback) [reply pandas/util/testing.py]: should try to completely remove these from testing at some point

**Comment by jreback** [comment]
thanks @jbrockmendel

**Pull request merged**